flamedisx/xenon/x1t_sr1.py

```diff
@@ -254,27 +282,85 @@ def set_defaults(self, *args, **kwargs):
+
+        return itp_bias + tf.ones_like(s2_raw, dtype=fd.float_type())
+
+    def reconstruction_smear_s2_simulate(self, s2_raw):
+        sig_tf = tf.convert_to_tensor(s2_raw, dtype=fd.float_type())
+        itp_smear = interpolate_tf(sig_tf,
+                self.recon_smear_s2_simulate_tf[0],
+                self.domain_def_smear_s2_simulate)
+        
+        return itp_smear+1e-45
+
+    # Backwards for annotate
+    # S1
+    def reconstruction_bias_s1_annotate(self, s1_raw):
+        # This function should return bias = reconstructed_area/raw_area
+        # Joran's map returns (reconstructed_area/raw_area - 1)
+        sig_tf = tf.convert_to_tensor(s1_raw, dtype=fd.float_type())
+        itp_bias = interpolate_tf(sig_tf,
+                self.recon_bias_s1_annotate_tf[0],
+                self.domain_def_bias_s1_annotate)
+
+        return itp_bias + tf.ones_like(s1_raw, dtype=fd.float_type())
+
+    def reconstruction_smear_s1_annotate(self, s1_raw):
+        sig_tf = tf.convert_to_tensor(s1_raw, dtype=fd.float_type())
+        itp_smear = interpolate_tf(sig_tf,
+                self.recon_smear_s1_annotate_tf[0],
+                self.domain_def_smear_s1_annotate)
+
+        return itp_smear+1e-45
+
+    # S2
+    def reconstruction_bias_s2_annotate(self, s2_raw):
+        # This function should return bias = reconstructed_area/raw_area
+        # Joran's map returns (reconstructed_area/raw_area - 1)
+        sig_tf = tf.convert_to_tensor(s2_raw, dtype=fd.float_type())
+        itp_bias = interpolate_tf(sig_tf,
+                self.recon_bias_s2_annotate_tf[0],
+                self.domain_def_bias_s2_annotate)
+
+        return itp_bias + tf.ones_like(s2_raw, dtype=fd.float_type())
+
+    def reconstruction_smear_s2_annotate(self, s2_raw):
+        sig_tf = tf.convert_to_tensor(s2_raw, dtype=fd.float_type())
+        itp_smear = interpolate_tf(sig_tf,
+                self.recon_smear_s2_annotate_tf[0],
+                self.domain_def_smear_s2_annotate)
+        
+        return itp_smear+1e-45
 
-    def reconstruction_bias_s1(self,
-                               s1,
-                               s1_reconstruction_bias_pivot=\
-                                   DEFAULT_S1_RECONSTRUCTION_BIAS_PIVOT):
-        return calculate_reconstruction_bias(
-            s1,
-            self.recon_map_s1_tf,
-            self.domain_def_s1,
-            pivot_pt=s1_reconstruction_bias_pivot)
-
-    def reconstruction_bias_s2(self,
-                               s2,
-                               s2_reconstruction_bias_pivot=\
-                                   DEFAULT_S2_RECONSTRUCTION_BIAS_PIVOT):
-        return calculate_reconstruction_bias(
-            s2,
-            self.recon_map_s2_tf,
-            self.domain_def_s2,
-            pivot_pt=s2_reconstruction_bias_pivot)
-
+    # Not sure what random truth is for
     def random_truth(self, n_events, fix_truth=None, **params):
```

Comment:
It's for simulating the truth (energy, position) of the event. Maybe remove this comment?

Reply:
Done